source/particle/integrator/euler.cc

```diff
@@ -40,7 +40,13 @@ namespace aspect
                           "to the number of particles to advect. For some unknown reason they are different, "
                           "most likely something went wrong in the calling function."));
 
-        const bool geometry_has_periodic_boundary = (this->get_geometry_model().get_periodic_boundary_pairs().size() != 0);
+        const auto &cell = begin_particle->get_surrounding_cell();
```

Comment:
I think I would prefer making `cell` not a reference:
```suggestion
        const auto cell = begin_particle->get_surrounding_cell();
```